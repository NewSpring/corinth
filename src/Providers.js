import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import { LiveProvider } from '@apollosproject/ui-connected';

import NavigationService from './NavigationService';
import ClientProvider from './client';
import ExternalLinkProvider from './linking/Provider';
import track from './analytics';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      navigate={NavigationService.navigate}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() => NavigationService.navigate('Onboarding')}
      >
        <ExternalLinkProvider navigate={NavigationService.navigate}>
          <MediaPlayerProvider>
            <AnalyticsProvider trackFunctions={[track]}>
              <LiveProvider>
                <Providers
                  themeInput={customTheme}
                  iconInput={customIcons}
                  {...props}
                />
              </LiveProvider>
            </AnalyticsProvider>
          </MediaPlayerProvider>
        </ExternalLinkProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
