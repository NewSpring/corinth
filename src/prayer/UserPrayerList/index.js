import React from 'react';
import { Mutation, Query } from 'react-apollo';
import { View, ScrollView, ActivityIndicator } from 'react-native';
import { SafeAreaView } from 'react-navigation';
import {
  Card,
  CardContent,
  BodyText,
  H4,
  H6,
  ModalView,
  PaddedView,
  styled,
  Icon,
  withTheme,
} from '@apollosproject/ui-kit';
import PrayerSingle from '../PrayerSingle';
import GET_USER_PRAYERS from '../data/queries/getUserPrayers';
import DELETE_PRAYER from '../data/mutations/deletePrayer';
import ActionComponent from '../ActionComponent';

const FlexedSafeAreaView = styled({
  flex: 1,
})(SafeAreaView);

const StyledView = styled(({ theme }) => ({
  marginBottom: theme.sizing.baseUnit * 4,
}))(View);

const GreenH4 = styled(({ theme }) => ({
  color: theme.colors.primary,
}))(H4);

const DeleteIcon = withTheme(({ theme }) => ({
  name: 'delete',
  size: theme.sizing.baseUnit,
  fill: theme.colors.darkTertiary,
}))(Icon);

class UserPrayerList extends React.Component {
  static navigationOptions = {
    header: null,
  };

  render() {
    return (
      <ModalView {...this.props} onClose={() => this.props.navigation.pop()}>
        <FlexedSafeAreaView forceInset={{ top: 'always' }}>
          <ScrollView>
            <Query query={GET_USER_PRAYERS} fetchPolicy="cache-and-network">
              {({ loading, data: { userPrayers = [] } = {} }) => {
                if (loading) return <ActivityIndicator />;
                return (
                  <>
                    <PaddedView>
                      <H6>Viewing</H6>
                      <GreenH4>My Prayers</GreenH4>
                    </PaddedView>
                    <Mutation
                      mutation={DELETE_PRAYER}
                      update={(cache, { data: { deletePrayer } }) => {
                        const data = cache.readQuery({
                          query: GET_USER_PRAYERS,
                        });
                        const { id } = deletePrayer;
                        const updatedPrayers = data.userPrayers.filter(
                          (prayer) => prayer.id !== id
                        );
                        cache.writeQuery({
                          query: GET_USER_PRAYERS,
                          data: { userPrayers: updatedPrayers },
                        });
                      }}
                    >
                      {(deletePrayer) => (
                        <StyledView>
                          {userPrayers && userPrayers.length > 0 ? (
                            userPrayers.map((prayer) => (
                              <Card key={prayer.id}>
                                <CardContent>
                                  <PrayerSingle
                                    prayer={prayer}
                                    showDate
                                    action={
                                      <ActionComponent
                                        component={<DeleteIcon />}
                                        options={[
                                          {
                                            title: 'Delete Prayer',
                                            method: async () => {
                                              await deletePrayer({
                                                variables: {
                                                  parsedId: prayer.id,
                                                },
                                              });
                                            },
                                            destructive: true,
                                          },
                                          {
                                            title: 'Cancel',
                                            method: null,
                                            destructive: false,
                                          },
                                        ]}
                                      />
                                    }
                                  />
                                </CardContent>
                              </Card>
                            ))
                          ) : (
                            <PaddedView>
                              <BodyText>
                                You have not submitted any prayers. Go back and
                                add one!
                              </BodyText>
                            </PaddedView>
                          )}
                        </StyledView>
                      )}
                    </Mutation>
                  </>
                );
              }}
            </Query>
          </ScrollView>
        </FlexedSafeAreaView>
      </ModalView>
    );
  }
}

export default UserPrayerList;
