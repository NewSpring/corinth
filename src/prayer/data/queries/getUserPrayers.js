import gql from 'graphql-tag';

export default gql`
  query UserPrayers {
    userPrayers {
      id
      text
      startTime
    }
  }
`;
