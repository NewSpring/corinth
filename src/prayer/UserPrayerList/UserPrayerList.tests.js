import React from 'react';
import Providers from 'newspringchurchapp/src/Providers';
import { renderWithApolloData } from 'newspringchurchapp/src/utils/testUtils';
import getUserPrayers from '../data/queries/getUserPrayers';
import UserPrayerList from '.';

const prayers = [
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4786',
    firstName: 'Aaron',
    lastName: 'Attendee',
    text: 'This is a test to see if I have a prayer request.',
    enteredDateTime: '2017-04-08T11:21:53.053',
  },
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4787',
    firstName: 'Aaron',
    lastName: 'Attendee',
    text: 'This is another test to see if I have a prayer request.',
    enteredDateTime: '2017-04-09T11:21:53.053',
  },
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4788',
    firstName: 'Aaron',
    lastName: 'Attendee',
    text: 'This is just another test to see if I have a prayer request.',
    enteredDateTime: '2017-04-11T11:21:53.053',
  },
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4789',
    firstName: 'Aaron',
    lastName: 'Attendee',
    text: 'This, yet again, is a test to see if I have a prayer request.',
    enteredDateTime: '2017-04-10T11:21:53.053',
  },
];

const mocks = [
  { request: { query: getUserPrayers }, response: { data: { prayers } } },
];

describe('the UserPrayerList component', () => {
  it('renders a list of prayers', async () => {
    const navigation = { pop: jest.fn() };
    const tree = await renderWithApolloData(
      <Providers mocks={mocks} addTypename={false}>
        <UserPrayerList navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders no prayer text', async () => {
    const navigation = { pop: jest.fn() };
    const tree = await renderWithApolloData(
      <Providers
        mocks={[
          {
            request: { query: getUserPrayers },
            response: { data: {} },
          },
        ]}
        addTypename={false}
      >
        <UserPrayerList navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
