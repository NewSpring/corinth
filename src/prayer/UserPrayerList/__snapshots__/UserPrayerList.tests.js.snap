// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the UserPrayerList component renders a list of prayers 1`] = `
<BVLinearGradient
  colors={
    Array [
      4294967295,
      4294506484,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "borderRadius": 0,
      "flex": null,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    forceInset={
      Object {
        "top": "always",
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#505050",
                "fontFamily": "Colfax-Bold",
                "fontSize": 12,
                "lineHeight": 17.28,
              }
            }
          >
            Viewing
          </Text>
          <Text
            style={
              Object {
                "color": "#6bac43",
                "fontFamily": "Colfax-Bold",
                "fontSize": 16,
                "lineHeight": 23.04,
              }
            }
          >
            My Prayers
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 64,
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 16,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              bold={false}
              italic={false}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "Colfax-Regular",
                  "fontSize": 16,
                  "lineHeight": 26,
                }
              }
            >
              You have not submitted any prayers. Go back and add one!
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 16,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={32}
      bbWidth={32}
      fill="#303030"
      height={32}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          null,
          Object {
            "flex": 0,
            "height": 32,
            "width": 32,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={32}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4281348144,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        borderRadius={24}
        iconPadding={8}
        style={
          Object {
            "backgroundColor": Object {
              "color": Array [
                48,
                48,
                48,
              ],
              "model": "rgb",
              "valpha": 0.7,
            },
            "padding": 8,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={16}
          bbWidth={16}
          height={16}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={16}
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGPath
              d="M13.07 12l7.2-7.2.5-.5c.3-.3.3-.78 0-1.08-.3-.3-.76-.3-1.06 0l-.5.5-7.2 7.2-7.2-7.2-.5-.5c-.3-.3-.77-.3-1.07 0-.3.3-.3.77 0 1.07l.5.5 7.2 7.2-7.7 7.7c-.3.3-.3.77 0 1.07.3.3.77.3 1.07 0l7.7-7.7 7.7 7.7c.3.3.78.3 1.08 0 .3-.3.3-.77 0-1.07l-7.7-7.7z"
              fill={
                Array [
                  0,
                  4294967295,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</BVLinearGradient>
`;

exports[`the UserPrayerList component renders no prayer text 1`] = `
<BVLinearGradient
  colors={
    Array [
      4294967295,
      4294506484,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "borderRadius": 0,
      "flex": null,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    forceInset={
      Object {
        "top": "always",
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#505050",
                "fontFamily": "Colfax-Bold",
                "fontSize": 12,
                "lineHeight": 17.28,
              }
            }
          >
            Viewing
          </Text>
          <Text
            style={
              Object {
                "color": "#6bac43",
                "fontFamily": "Colfax-Bold",
                "fontSize": 16,
                "lineHeight": 23.04,
              }
            }
          >
            My Prayers
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 64,
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 16,
                "paddingVertical": 16,
              }
            }
          >
            <Text
              bold={false}
              italic={false}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "Colfax-Regular",
                  "fontSize": 16,
                  "lineHeight": 26,
                }
              }
            >
              You have not submitted any prayers. Go back and add one!
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 16,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={32}
      bbWidth={32}
      fill="#303030"
      height={32}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          null,
          Object {
            "flex": 0,
            "height": 32,
            "width": 32,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={32}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4281348144,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        borderRadius={24}
        iconPadding={8}
        style={
          Object {
            "backgroundColor": Object {
              "color": Array [
                48,
                48,
                48,
              ],
              "model": "rgb",
              "valpha": 0.7,
            },
            "padding": 8,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={16}
          bbWidth={16}
          height={16}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              undefined,
              Object {
                "opacity": 1,
              },
              Object {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={16}
        >
          <RNSVGGroup
            fill={
              Array [
                0,
                4278190080,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            font={Object {}}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={Array []}
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth={1}
            vectorEffect={0}
            x={0}
            y={0}
          >
            <RNSVGPath
              d="M13.07 12l7.2-7.2.5-.5c.3-.3.3-.78 0-1.08-.3-.3-.76-.3-1.06 0l-.5.5-7.2 7.2-7.2-7.2-.5-.5c-.3-.3-.77-.3-1.07 0-.3.3-.3.77 0 1.07l.5.5 7.2 7.2-7.7 7.7c-.3.3-.3.77 0 1.07.3.3.77.3 1.07 0l7.7-7.7 7.7 7.7c.3.3.78.3 1.08 0 .3-.3.3-.77 0-1.07l-7.7-7.7z"
              fill={
                Array [
                  0,
                  4294967295,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</BVLinearGradient>
`;
